app/backend/src/couchers/servicers/admin.py

```diff
@@ -288,6 +293,87 @@ def CreateCommunity(self, request, context, session):
 
         return community_to_pb(session, node, context)
 
+    def UpdateCommunity(self, request, context, session):
+        cluster = session.execute(
+            select(Cluster).where(Cluster.id == request.community_id).where(Cluster.deleted == None)
+        ).scalar_one_or_none()
+        if not cluster:
+            context.abort(grpc.StatusCode.NOT_FOUND, errors.COMMUNITY_NOT_FOUND)
+
+        node = session.execute(select(Node).where(Node.id == cluster.parent_node_id)).scalar_one_or_none()
+        if not node:
+            context.abort(grpc.StatusCode.NOT_FOUND, errors.COMMUNITY_NOT_FOUND)
+
+        name = request.name if request.name else cluster.name
+        description = request.description if request.description else cluster.description
+
+        geom = node.geom
+        if request.geojson:
+            geom = shape(json.loads(request.geojson))
+            if geom.geom_type != "MultiPolygon":
+                context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.NO_MULTIPOLYGON)
+            geom = from_shape(geom)
+
+        if request.parent_node_id != 0 and request.parent_node_id != node.parent_node_id:
+            parent_node_id = request.parent_node_id
+
+            new_node = Node(geom=geom, parent_node_id=parent_node_id)
+            session.add(new_node)
+
+            now = datetime.utcnow()
+            session.execute(update(Node).where(Node.id == node.id).values({"deleted": now}))
```

Comment:
It's better to do this through the ORM, you can simply assign to `node.deleted`, so this would be

```py
node.deleted = now()
```

`session.execute` will actually issue a SQL statement here. Using the ORM way will issue them all at `flush` or `commit` (which happens automatically as well at the end of the RPC due to the `session_scope` context manager).

This applies to a bunch of places here. We only really need a low-level `UPDATE` when updating a lot of things (where it's undesirable to have SQLAlchemy load everything into python objects), or when updating based on a computed SQL expression we want rendered directly (e.g. `UPDATE table SET somevalue = othervalue + 2...` or similar).

I am also not entirely sure how this interacts with the SQLAlchemy ORM: it might cause a bunch of refreshing objects (extra `SELECT`s), or it might cause the object to be stale in other ways.